x/vmbridge/keeper/evm.go

```diff
@@ -0,0 +1,134 @@
+package keeper
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	ethermint "github.com/okex/exchain/app/types"
+	sdk "github.com/okex/exchain/libs/cosmos-sdk/types"
+	sdkerrors "github.com/okex/exchain/libs/cosmos-sdk/types/errors"
+	tmtypes "github.com/okex/exchain/libs/tendermint/types"
+	erc20types "github.com/okex/exchain/x/erc20/types"
+	evmtypes "github.com/okex/exchain/x/evm/types"
+	"github.com/okex/exchain/x/vmbridge/types"
+	"math/big"
+)
+
+// event __SendToWasmEventName(string wasmAddr,string recipient, string amount)
+type SendToWasmEventHandler struct {
+	Keeper
+}
+
+func NewSendToWasmEventHandler(k Keeper) *SendToWasmEventHandler {
+	return &SendToWasmEventHandler{k}
+}
+
+// EventID Return the id of the log signature it handles
+func (h SendToWasmEventHandler) EventID() common.Hash {
+	return types.SendToWasmEvent.ID
+}
+
+// Handle Process the log
+func (h SendToWasmEventHandler) Handle(ctx sdk.Context, contract common.Address, data []byte) error {
+	if !tmtypes.HigherThanEarth(ctx.BlockHeight()) {
+		errMsg := fmt.Sprintf("vmbridger not supprt at height %d", ctx.BlockHeight())
+		return sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
+	}
+
+	logger := h.Keeper.Logger()
+	unpacked, err := types.SendToWasmEvent.Inputs.Unpack(data)
```

Comment:
is there any condition that the length of `unpacked` is not 3 or the filed type of it if not `string, string, *big,Int`? Maybe it is better to check it and add some annotations

Reply:
It's nice suggestion，i will add more unit test to check it. But Input.Unpack is ETH abi interface of go language, it's risk not high

Reply:
I add some unit test. fake data ( bytes,string,big.int) it can be normal unpack, but after will check
 address. And fake data (string,string,int8(-1)) it also can be normal unpack, but is panic in sdk.NewIntFromBigInt  it's correct action. 
The most importante case in func (lh LogProcessEvmHook) PostTxProcessing it will search event.ID to handle different logic. event.ID = hash(event.names, events.inputs.type)。So we can regard Unpack.data is must be correct type。